src/completions/Tab.ts

```diff
@@ -238,4 +240,32 @@ export class BufferCompletionSource extends Completions.CompletionSourceFuse {
         }
         return this.updateDisplay()
     }
+
+    /**
+     * Update the list of possible tab options and select (focus on)
```

Comment:
This makes me uneasy, I think it could cause confusion.

I would store the current options, update the options, then do a diff between them to see if the focused option was deleted, then focus the one after (I think this makes more sense as you are pressing delete and not backspace). If the option was not deleted focus should be kept on it, even if it has moved.

Ramda is probably a useful library for this; we use it in a few other places.

If that's too much work I'm happy to look at it myself :)

Reply:
Sounds like a plan - I can make these changes.

Reply:
Sorry for any inconvenience with the delay here - I pushed an update to address the above concerns.